src/MetricsUploader/include/MetricsUploader/ScopedMetric.h

```diff
@@ -23,11 +24,17 @@ class ScopedMetric {
 
   void SetStatusCode(OrbitLogEvent::StatusCode status_code) { status_code_ = status_code; }
 
+  void Pause();
+  void Resume();
+
  private:
   MetricsUploader* uploader_;
-  OrbitLogEvent_LogEventType log_event_type_;
+  OrbitLogEvent::LogEventType log_event_type_;
   OrbitLogEvent::StatusCode status_code_ = OrbitLogEvent::SUCCESS;
   std::chrono::steady_clock::time_point start_;
+
+  std::optional<std::chrono::steady_clock::time_point> pause_start_ = std::nullopt;
+  std::chrono::milliseconds pause_duration_ = std::chrono::milliseconds::zero();
```

Comment:
Just as an idea. It's up to you if you want to do that. You don't necessarily need the field `pause_duration_`. You could also change `start_` in the `Resume()` function. Basically like this: 
```
std::chrono::milliseconds pause_duration = std::chrono::duration_cast<std::chrono::milliseconds>(
      std::chrono::steady_clock::now() - pause_start_.value());
start_ += pause_duration;
pause_start_.reset();
```

Reply:
Thanks for the suggestion!

Agree that we can adjust the `start_`, but this makes the meaning of `start_` not very clear. It does not mean the start time of the metric any more.

Reply:
Yes. That is why I am also hesitant. Both options have their up and down sides. One could make the argument that `start_` is completely an implementation detail and that its not part of the public interface, so its okay to miss use it. (It could also be rename to something like `timer_start_point_`, but I dont know if that makes things better). On the other hand, its also easier to read / understand with the `pause_duration_`. Up to you :)